### PR TITLE
annotat unused vars

### PR DESCRIPTION
**Pull request by Borda**
## What does this PR do?

annotate all unused vars, need to be verified that is need for the API
if not needed replace all `_`

## Before submitting
- [x] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together? Otherwise, we ask you to create a separate PR for every change.
- [x] Did you make sure to update the documentation with your changes?
- [x] Did you write any new necessary tests? 
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)?

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:  

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified; _Bugfixes should be including in bug-fix release milestones (m.f.X) and features should be included in (m.X.b) releases._
 

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Review by tchaton — DISMISSED** [review]
LGTM !

File: pytorch_lightning/metrics/functional/f_beta.py [hunk 1]
```diff
@@ -12,6 +12,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 from typing import Tuple
+from warnings import warn
```
Comment (carmocca) [inline pytorch_lightning/metrics/functional/f_beta.py:15]: no `rank_zero_warn`?

File: pytorch_lightning/metrics/functional/psnr.py [hunk 2]
```diff
@@ -1,5 +1,6 @@
 
 from typing import Tuple, Optional
+from warnings import warn
```
Comment (carmocca) [inline pytorch_lightning/metrics/functional/psnr.py:3]: same here

**Comment by carmocca** [comment]
What is the plan with these TODOs? Fix them one by one?

**Comment by Borda** [comment]
> What is the plan with these TODOs? Fix them one by one?

exactly, this was the identification and now after 1.1 we shall fix them all, eventually write them as issues for new contributions

**Comment by carmocca** [comment]
> exactly, this was the identification and now after 1.1 we shall fix them all, eventually write them as issues for new contributions

Sounds good, but not all changes in this PR have their corresponding `TODO` comment. Might get lost later.

It might be a good idea to have it annotated as `TODO(#5017): bla bla bla` to have more context

File: pytorch_lightning/trainer/evaluation_loop.py [hunk 3]
```diff
@@ -90,14 +90,14 @@ def on_evaluation_start(self, *args, **kwargs):
         else:
             self.trainer.call_hook('on_validation_start', *args, **kwargs)
 
-    def on_evaluation_model_eval(self, *args, **kwargs):
+    def on_evaluation_model_eval(self, *_, **__):
```
Comment (awaelchli) [inline pytorch_lightning/trainer/evaluation_loop.py:93]: I've never seen this. Is this a new convention? :)

Comment (Borda) [reply pytorch_lightning/trainer/evaluation_loop.py:93]: I saw it already in our codebase https://github.com/PyTorchLightning/pytorch-lightning/blob/ce9179591dc3414aa0ff59007ea46f30fa9ddbee/pytorch_lightning/trainer/connectors/logger_connector/epoch_result_store.py#L140

**Comment by Borda** [comment]
> > exactly, this was the identification and now after 1.1 we shall fix them all, eventually write them as issues for new contributions
> 
> Sounds good, but not all changes in this PR have their corresponding `TODO` comment. Might get lost later.
> 
> It might be a good idea to have it annotated as `TODO(#5017): bla bla bla` to have more context

yes, agree but atm we do not have created all issue and also not clear how each shall be solved...

File: pytorch_lightning/trainer/evaluation_loop.py (hunk 3 above)
Comment (carmocca) [reply pytorch_lightning/trainer/evaluation_loop.py:93]: Definitely not new, here is a topic about it from 8 years ago :laughing: https://stackoverflow.com/a/11486174

**Comment by carmocca** [comment]
> yes, agree but atm we do not have created all issue and also not clear how each shall be solved...

I meant referencing this PR, not the number of each issue

File: pytorch_lightning/trainer/training_loop.py [hunk 4]
```diff
@@ -646,7 +646,7 @@ def run_training_batch(self, batch, batch_idx, dataloader_idx):
         grad_norm_dic = {}
 
         # bookkeeping
-        using_results_obj = False
+        # using_results_obj = False
```
Comment (carmocca) [inline pytorch_lightning/trainer/training_loop.py:649]: Add TODO comment or simply remove the line

**Review by SeanNaren — DISMISSED** [review]
Just make sure to get to @carmocca comments :)

**Review by justusschock — APPROVED** [review]

File: pytorch_lightning/distributed/dist.py [hunk 5]
```diff
@@ -48,9 +48,9 @@ def _emit(self, obj: Any, group=WORLD):
         torch.save(obj, buffer)
         data = bytearray(buffer.getbuffer())
         length_tensor = torch.tensor([len(data)]).long().to(self.device)
-        length_tensor = self._broadcast(length_tensor, src=0, group=group)
+        _ = self._broadcast(length_tensor, src=0, group=group)
```
Comment (Borda) [inline pytorch_lightning/distributed/dist.py:51]:
```suggestion
        self._broadcast(length_tensor, src=0, group=group)
```

File: pytorch_lightning/distributed/dist.py [hunk 6]
```diff
@@ -48,9 +48,9 @@ def _emit(self, obj: Any, group=WORLD):
...
         data_tensor = torch.ByteTensor(data).to(self.device)
-        data_tensor = self._broadcast(data_tensor, src=0, group=group)
+        _ = self._broadcast(data_tensor, src=0, group=group)
```
Comment (Borda) [inline pytorch_lightning/distributed/dist.py:53]:
```suggestion
        self._broadcast(data_tensor, src=0, group=group)
```

**Review by carmocca — APPROVED** [review]

**Review by tchaton — APPROVED** [review]

**Review by justusschock — APPROVED** [review]

**Review by carmocca — APPROVED** [review]

**Review by SeanNaren — APPROVED** [review]

**Comment by Borda** [comment]
@williamFalcon mind review, it just adding todos to feature branch, no master...

**Review by tchaton — APPROVED** [review]
